packages/ui/src/theme/__tests__/findDeprecatedStateTokens.test.ts

```diff
@@ -0,0 +1,55 @@
+import { findDeprecatedStateTokens } from '../utils';
+
+describe('@aws-amplify/ui', () => {
+  describe('findDeprecatedStateTokens', () => {
+    let baseTheme;
+    beforeEach(() => {
+      baseTheme = {
+        name: 'baseTheme',
+        tokens: {
+          components: {
+            button: {
+              hover: {
+                color: { value: 'red', deprecatedStateToken: true },
+                backgroundColor: { value: 'blue' },
+              },
+              focus: {
+                fontSize: {
+                  value: 'fontSizes.xxxl',
+                  deprecatedStateToken: true,
+                },
+              },
+            },
+          },
+        },
+      };
+    });
+
+    it('should return a list of pathObjects', () => {
+      const pathObjects = findDeprecatedStateTokens(baseTheme);
+      const expectedPaths = [
+        ['components', 'button', 'hover', 'color'],
+        ['components', 'button', 'focus', 'fontSize'],
+      ];
+      expect(pathObjects.length).toEqual(2);
+      expectedPaths.forEach((path) => {
+        const containsExpected = pathObjects.some((pathObj) => {
+          return (
+            pathObj.path.length === path.length &&
+            path.every((val, idx) => val === pathObj.path[idx])
```

Comment:
Seems like a lot of nested iterations (`forEach`, `some`, `every`). Perhaps there's a more efficient way to do this?

Reply:
The reason I'm doing it like this is so that the test is just reliant on the output and not the ordering of it.  And since this is just a test file the data set is small and know so I think the nested iteration is acceptable, but this could also be updated to just compare the first result value with the first expected value and skip the nested iteration but then if for whatever reason the ordering of the results every changed the test would fail even though the current ordering isn't a requirement for the algorithm to be correct.